File: Java Programming/problems on file handling/problem15.java
/*
15.  Write a program which accept file name which contains information of student as well as accept 
		name of student and display only information of that student. 	
	
	  Author : Milind Nalawade
	
	  Date : 20/08/2021

      Language : Java 14.0.2
*/
import java.lang.*;
import java.util.*;
import java.io.Serializable;
import java.io.FileOutputStream;
import java.io.IOException;
import java.io.ObjectOutputStream;
import java.io.ObjectInputStream;
import java.io.FileInputStream;
import java.lang.Exception;
class Info implements Serializable
{
	private String name;
	private int id;
	private int marks;

	public void Name(String n)
	{
		this.name = n;
	}
	public String getName()
	{
		return name;
	}

	public int getid() {
		return id;
	}

	public void setid(int ID) {
		this.id = ID;
	}

	public int getmarks() {
		return marks;
	}

	public void setmarks(int Marks) {
		this.marks = Marks;
	}

	@Override
	public String toString()
	{
		return " Name=" + this.name ;
	}

}



class Student
{
	static void insert(String filename,ArrayList<Info>ali)
	{
		try
		{
			FileOutputStream fos=new FileOutputStream(filename);
			if(fos==null)
			{
				System.out.print("\n Unable to open file \n");
			}
			else
			{
				System.out.print("\n File successfully opened\n");
			}
			ObjectOutputStream oos=new ObjectOutputStream(fos);
			oos.writeObject(ali);
			System.out.print("\nDone\n");
			
		}
		catch(IOException e)
		{
			e.printStackTrace();
		}
	}
	
	static void Display(String filename)
	{
		int sum=0;
		Scanner sobj=new Scanner(System.in);
		try
		{
			FileInputStream fis=new FileInputStream(filename);
			ObjectInputStream ois=new ObjectInputStream(fis);
			Info iobj=null;
			ArrayList<Info> ali=new ArrayList<>();
			ali=(ArrayList<Info>)ois.readObject();
			System.out.print("\n Enter name of student : ");
			String str=sobj.nextLine();
			for(int i=0;i<ali.size();i++)
			{
				String arr=ali.get(i).getName();
				if(str.compareTo(arr)==0)
				{
					System.out.print("\n Name : "+ali.get(i).getName()+"\n ID : "+ali.get(i).getid()+"\n Marks : "+ali.get(i).getmarks());
				}
			}
		}
		catch (Exception e)
		{
			
		}		
	}
}

class problem15
{
	public static void main(String argv[])
	{
		Scanner sobj=new Scanner(System.in);
		System.out.print("\n Enter file name : ");
		String filename=sobj.nextLine();
		Student obj=new Student();
		System.out.print("\n Enter number of students : ");
		int size=sobj.nextInt();
		ArrayList<Info> ali=new ArrayList<>();
		for(int i=0;i<size;i++)
		{
			System.out.println();
			System.out.print(" Enter name of student : ");
			sobj.nextLine();
			String name=sobj.nextLine();
			System.out.println();
			System.out.print(" Enter id of student : ");
			int ID=sobj.nextInt();
			System.out.println();
			System.out.print(" Enter marks of student : ");
			int Marks=sobj.nextInt();
			Info iobj=new Info();
			iobj.Name(name);
			iobj.setid(ID);
			iobj.setmarks(Marks);
			ali.add(iobj);
		}
		//obj.insert(filename,ali);
		obj.Display(filename);
	}
}

File: C Programming/problems on file handling/problem15.c
/*
15.  Write a program which accept file name which contains information of student as well as accept 
		name of student and display only information of that student. 	
	
	   Author : Milind Nalawade
	
	  Date : 20/08/2021
	
      Language : C programming
	
*/

#include<stdio.h>
#include<stdlib.h>
#include<unistd.h>
#include<fcntl.h>
#include<string.h>
struct demo
{
	char name[20];
	int id;
	int marks;
};

void insert(char*name)
{
	struct demo obj;
	int fd=0,ret=0,i=0,size=0;
	fd=open(name,O_WRONLY);
	if(fd==-1)
	{
	printf("\n unable ot open the file\n");
	return;
	}
	else
	{
		printf("\n File successfully opened \n");
	}
	printf("\n Enter the number of students : ");
	scanf("%d",&size);
	printf("\n");
	for(i=0;i<size;i++)
	{
		printf(" Enter the name of student :");
		scanf(" %[^'\n']c",obj.name);
		printf(" Enter the id of student : ");
		scanf("%d",&obj.id);
		printf(" Enter the marks of student :");
		scanf("%d",&obj.marks);
		printf("\n");
		write(fd,&obj,sizeof(obj));
		memset(&obj,0,sizeof(obj));
	}
	close(fd);
}

void display(char*name)
{
	struct demo obj;
	int j=0,ret=0,fd=0;
	char arr[20]={'\0'};
	fd=open(name,O_RDONLY);
	if(fd==-1)
	{
		printf("\n Unable to open the the file\n");
		return ;	
	}
	else
	{
		printf("\n File successfully gets opened\n");
	}

	printf("\n The name of the students : ");
	scanf("%s",&arr);
	while((ret=read(fd,&obj,sizeof(obj)))!=0)
	{
		if(strcmp(arr,obj.name)==0)
		{
			printf("\n name : %s\n Id : %d\n marks : %d \n",obj.name,obj.id,obj.marks);
		}
	}
	close(fd);
}

int main()
{
	char name[20]={'\0'};
	printf("\n Enter the file name :");
	scanf("%s",name);
	//insert(name);
	display(name);
	return 0;
}
